internal/keepers/ocr.go

```diff
@@ -94,7 +102,13 @@ func (k *keepers) Report(ctx context.Context, rt types.ReportTimestamp, _ types.
 
 	// pass the filter to the dedupe function
 	// ensure no locked keys come through
-	keys, err := shuffledDedupedKeyList(attributed, key, k.filter.Filter())
+	keys, err := shuffledDedupedKeyList(
+		attributed,
+		key,
+		observationKeysLimit,
+		reportKeysLimit,
```

Comment:
again, i think we are really expanding the scope of shuffle and deduping. having separate steps would be clearer

Reply:
It would be more tricky since we need to apply these limits in the middle of that func, and in the middle of the loop.

Reply:
@ferglor is also doing changes in this function which will refactor and break it into multiple pieces. I think the observation limit will then fit nicely there. 

let's wait for his changes and then we can rebase this PR on top of that